src/v/raft/vote_stm.cc

```diff
@@ -192,14 +191,65 @@ ss::future<election_success> vote_stm::do_vote() {
     co_return election_success(_success);
 }
 
+bool vote_stm::has_request_in_progress() const {
+    return std::any_of(
+      _replies.begin(),
+      _replies.end(),
+      [](const absl::flat_hash_map<vnode, vmeta>::value_type& p) {
+          return p.second.get_state() == vmeta::state::in_progress;
+      });
+}
+
+bool vote_stm::can_wait_for_all() const {
+    return clock_type::now()
+           < _requests_dispatched_ts + _ptr->_jit.base_duration();
+}
+
 ss::future<> vote_stm::process_replies() {
     return ss::repeat([this] {
+        const bool request_in_progress = has_request_in_progress();
+
+        /**
+         * Try to wait for vote replies from all of the nodes or use information
+         * from all replies if it is already available. Waiting for all the
+         * replies allow candidate to verify if any other protocol participant
+         * has log which is longer.
+         *
+         * If any of the replicas log is longer than current candidate the
+         * election is failed allowing the other node to become a leader.
+         */
+        if (
+          (!request_in_progress || can_wait_for_all())
+          && _ptr->_enable_longest_log_detection()) {
+            if (request_in_progress) {
+                return wait_for_next_reply().then(
```

Comment:
should `wait_for_next_reply` signal that it failed due to a timeout so that we can exit the loop?

Reply:
we chatted about that offline. Addeed an appropriate comment.